src/wallet/rpcdump.cpp

```diff
@@ -265,6 +297,33 @@ Value importwallet(const Array& params, bool fHelp)
         boost::split(vstr, line, boost::is_any_of(" "));
         if (vstr.size() < 2)
             continue;
+
+        // Let's see if the address is a valid Zcash spending key
+        if (fImportZKeys) {
+            try {
+                CZCSpendingKey spendingkey(vstr[0]);
+                libzcash::SpendingKey key = spendingkey.Get();
+                libzcash::PaymentAddress addr = key.address();
+                if (pwalletMain->HaveSpendingKey(addr)) {
+                    LogPrintf("Skipping import of zaddr %s (key already present)\n", CZCPaymentAddress(addr).ToString());
+                    continue;
+                }
+                int64_t nTime = DecodeDumpTime(vstr[1]);
+                LogPrintf("Importing zaddr %s...\n", CZCPaymentAddress(addr).ToString());
+                if (!pwalletMain->AddZKey(key)) {
+                    // Something went wrong
+                    fGood = false;
+                    continue;
```

Comment:
Are we relying on the ``CZCPaymentAddress`` constructor to throw so that this doesn't bypass the processing of Bitcoin addresses after the catch clause? That seems ugly; I would have expected that we check for a Zcash key prefix and perform the Zcash-specific processing based on that. This way there's no way to report errors from processing of keys that have the Zcash prefix but are invalid.